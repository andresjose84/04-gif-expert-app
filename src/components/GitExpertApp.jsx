import React, { useState } from 'react';
import AddCategory from './AddCategory';
import GifGrid from './GifGrid';

const GitExpertApp = ({ defaultCategories = [] }) => {

    //const categories = ['One punch','Samurai X', 'Dragon Ball'];
    const [categories, setCategories] = useState(defaultCategories);

    return (<>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={setCategories}></AddCategory>
        <hr />
        <ol>
            { categories.map((ele,key) => (
                <GifGrid key = { ele } category={ ele } indx = {key} />
            ) ) }
        </ol>
    </>)
};

export default GitExpertApp;