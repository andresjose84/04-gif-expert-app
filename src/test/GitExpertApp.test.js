import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GitExpertApp from '../components/GitExpertApp';

describe('Pruebas <GitExpertApp />', () => { 

    test('debe de crear un snapshot', () => { 
        const wrapper = shallow( <GitExpertApp /> );
        expect(wrapper).toMatchSnapshot();  
    });

    test('debe de mostrar una lista de categorias', () => { 
        const categories = ['Dragon ball','Ultrainstinto'];
        const wrapper = shallow( <GitExpertApp defaultCategories={categories}/> );
        expect(wrapper).toMatchSnapshot();  
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });

});