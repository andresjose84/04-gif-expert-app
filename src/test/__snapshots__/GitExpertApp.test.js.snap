// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GitExpertApp /> debe de crear un snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas <GitExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon ball"
      indx={0}
      key="Dragon ball"
    />
    <GifGrid
      category="Ultrainstinto"
      indx={1}
      key="Ultrainstinto"
    />
  </ol>
</Fragment>
`;
